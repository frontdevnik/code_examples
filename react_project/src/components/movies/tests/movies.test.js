import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Movies from '../Movies';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('<Movies />', () => {
  let wrapper;
  const mockStore = configureStore();
  const initialState = {
    fetchMovies: jest.fn(),
    fetchActors: jest.fn(),
    removeMovieReducer: jest.fn(),
    moviesReducer: {
      movies: [{
        title: 'Title',
        posterUrl: 'posterUrl',
        stars: '4',
        likes: '3',
        genres: ['Genres'],
        director: 'Director',
        actorsIds: [1, 2, 3],
        description: 'Description',
        id: 1,
      }],
      loading: false,
      error: null,
    },
  };

  beforeEach(() => {
    const store = mockStore(initialState);
    wrapper = shallow(<Movies store={store} />);
  });

  it('should render loading', () => {
    const initialState = {
      fetchMovies: jest.fn(),
      fetchActors: jest.fn(),
      removeMovieReducer: jest.fn(),
      moviesReducer: {
        movies: null,
        loading: true,
        error: null,
      },
    };
    const store = mockStore(initialState);
    wrapper = shallow(<Movies store={store} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render error', () => {
    const initialState = {
      fetchMovies: jest.fn(),
      fetchActors: jest.fn(),
      removeMovieReducer: jest.fn(),
      moviesReducer: {
        movies: [],
        loading: false,
        error: 'error',
      },
    };
    const store = mockStore(initialState);
    wrapper = shallow(<Movies store={store} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(wrapper.dive().dive().debug()).toMatchSnapshot();
  });
});
