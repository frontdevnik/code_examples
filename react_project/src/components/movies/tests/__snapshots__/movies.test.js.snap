// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies /> click on title should redirect 1`] = `
"<section className=\\"movies\\">
  <article className=\\"movie\\">
    <img className=\\"img\\" src=\\"posterUrl\\" alt=\\"Title\\" />
    <h2 className=\\"title\\" onClick={[Function: onClick]}>
      Title
    </h2>
    <Connect(Likes) likes=\\"3\\" id={1} movie={{...}} />
    <RatingStart stars=\\"4\\" id={1} movie={{...}} />
  </article>
</section>"
`;

exports[`<Movies /> should render correctly 1`] = `
"<section className=\\"movies\\">
  <article className=\\"movie\\">
    <img className=\\"img\\" src=\\"posterUrl\\" alt=\\"Title\\" />
    <h2 className=\\"title\\" onClick={[Function: onClick]}>
      Title
    </h2>
    <Connect(Likes) likes=\\"3\\" id={1} movie={{...}} />
    <RatingStart stars=\\"4\\" id={1} movie={{...}} />
  </article>
</section>"
`;

exports[`<Movies /> should render correctly 2`] = `
"<section className=\\"movies\\">
  <article className=\\"movie\\">
    <img className=\\"img\\" src=\\"posterUrl\\" alt=\\"Title\\" />
    <h2 className=\\"title\\" onClick={[Function: onClick]}>
      Title
    </h2>
    <Connect(Likes) likes=\\"3\\" id={1} movie={{...}} />
    <RatingStart stars=\\"4\\" id={1} movie={{...}} />
  </article>
</section>"
`;

exports[`<Movies /> should render error 1`] = `"<h1 style=\\"text-align:center\\">Sorry, the service is not available now, try visiting later</h1>"`;

exports[`<Movies /> should render loading 1`] = `"<div class=\\"loader\\"><span>Loading...</span></div>"`;

exports[`<Movies /> title on click should redirect 1`] = `
"<section className=\\"movies\\">
  <article className=\\"movie\\">
    <img className=\\"img\\" src=\\"posterUrl\\" alt=\\"Title\\" />
    <h2 className=\\"title\\" onClick={[Function: onClick]}>
      Title
    </h2>
    <Connect(Likes) likes=\\"3\\" id={1} movie={{...}} />
    <RatingStart stars=\\"4\\" id={1} movie={{...}} />
  </article>
</section>"
`;
